garage/tf/algos/reps.py

```diff
@@ -0,0 +1,419 @@
+        dual_before = self.f_dual(*dual_opt_input_values)
+        logger.log("Optimizing dual")
+
+        def eval_dual(x):
+            self.param_eta = x[0]
+            self.param_v = x[1:]
+            dual_opt_input_values = self._dual_opt_input_values(samples_data)
+            return self.f_dual(*dual_opt_input_values)
+
+        def eval_dual_grad(x):
+            self.param_eta = x[0]
+            self.param_v = x[1:]
+            dual_opt_input_values = self._dual_opt_input_values(samples_data)
+            grad = self.f_dual_grad(*dual_opt_input_values)
+            eta_grad = np.float(grad[0])
+            v_grad = grad[1]
+            return np.hstack([eta_grad, v_grad])
+
+        params_ast, _, _ = self.dual_optimizer(
+            func=eval_dual,
+            x0=x0,
+            fprime=eval_dual_grad,
+            bounds=bounds,
+            **self.dual_optimizer_args,
+        )
+
+        logger.log("Computing dual after")
+        self.param_eta, self.param_v = params_ast[0], params_ast[1:]
+        dual_opt_input_values = self._dual_opt_input_values(samples_data)
+        dual_after = self.f_dual(*dual_opt_input_values)
+
+        # Optimize policy
+        policy_opt_input_values = self._policy_opt_input_values(samples_data)
+        logger.log("Computing policy loss before")
+        loss_before = self.optimizer.loss(policy_opt_input_values)
+        logger.log("Computing policy KL before")
+        policy_kl_before = self.f_policy_kl(*policy_opt_input_values)
+        logger.log("Optimizing policy")
+        self.optimizer.optimize(policy_opt_input_values)
+        logger.log("Computing policy KL")
+        policy_kl = self.f_policy_kl(*policy_opt_input_values)
+        logger.log("Computing policy loss after")
+        loss_after = self.optimizer.loss(policy_opt_input_values)
+        logger.record_tabular("EtaBefore", eta_before)
+        logger.record_tabular("EtaAfter", self.param_eta)
+        logger.record_tabular("DualBefore", dual_before)
+        logger.record_tabular("DualAfter", dual_after)
+        logger.record_tabular("{}/LossBefore".format(self.policy.name),
+                              loss_before)
+        logger.record_tabular("{}/LossAfter".format(self.policy.name),
+                              loss_after)
+        logger.record_tabular("{}/dLoss".format(self.policy.name),
+                              loss_before - loss_after)
+        logger.record_tabular("{}/KLBefore".format(self.policy.name),
+                              policy_kl_before)
+        logger.record_tabular("{}/KL".format(self.policy.name), policy_kl)
+
+    def _build_inputs(self):
+        """Decalre graph inputs variables."""
+        observation_space = self.policy.observation_space
+        action_space = self.policy.action_space
+        policy_dist = self.policy.distribution
+
+        with tf.name_scope("inputs"):
+            obs_var = observation_space.new_tensor_variable(
+                name="obs", extra_dims=2)
```

Comment:
i know YAPF won't force you to, but this block is probably more readable if each of these arguments are split into different lines

```python
valid_var = tensor_utils.new_tensor(
    name="valid",
    ndim=2,
    dtype=tf.float32)
```
...or similar

Reply:
em, but YAPF will force me to split into different lines
```
--- garage/tf/algos/reps.py	(original)
+++ garage/tf/algos/reps.py	(reformatted)
@@ -163,36 +163,22 @@
 
         with tf.name_scope("inputs"):
             obs_var = observation_space.new_tensor_variable(
-                name="obs",
-                extra_dims=2)
+                name="obs", extra_dims=2)

```

Reply:
oh i see

i think you can do 
```python
valid_var = tensor_utils.new_tensor(
    name="valid",
    ndim=2,
    dtype=tf.float32)  # yapf: disable
```